src/corona_hakab_model/supervisor.py

```diff
@@ -269,3 +275,21 @@ def get(self, manager) -> float:
 
     def name(self) -> str:
         return "Total(" + ", ".join(n.name() for n in self.inners)
+
+class _EffectiveR0Supervisable (FloatSupervisable):
+    def __init__(self):
+        super().__init__()
+
+    def get(self, manager) -> float:
+        # note that this calculation is VARY heavy (that's why it is only calculating once every 5 days)
+        suseptable_indexes = [index for index, val in enumerate(manager.susceptible_vector) if val]
+        return np.sum(1 - np.exp(item) for item in manager.matrix.matrix[suseptable_indexes, :].data) * manager.matrix.total_contagious_probability / manager.matrix.size
+
+    def snapshot(self, manager):
+        if manager.current_date % 5 != 0 and not manager.current_date <= 1:
+            return
```

Comment:
snapsot shouldn't be overriden, make this check in get() (and return numpy.nan)

Reply:
returning numpy.nan destroys the plot (it doesnt know where to place those points so doesnt print at all).
for now i just removed this part, since with the generator removal, it runs fast enough to calculate at every turn.
however, i don't really see any problem with overriding the snapshot.
another possible solution is to update the base snapshot so that it won't add any point that is np.nan

Reply:
Delay already adds NANs, you can do it, it's fine